src/main.rs

```diff
@@ -80,12 +84,26 @@ fn main() -> Result<()> {
             log::warn!("Short delay: release any keys now!");
             std::thread::sleep(Duration::from_secs_f64(delay));
 
-            let device_info = deviceinfo::DeviceInfo::with_name(
-                &mapping_config.device_name,
-                mapping_config.phys.as_deref(),
-            )?;
+            let mut device_info;
+            let mut i = 0f64;
+            loop {
+                device_info = deviceinfo::DeviceInfo::with_name(
+                    &mapping_config.device_name,
+                    mapping_config.phys.as_deref(),
+                );
+                if !wait || device_info.is_ok() {
+                    break;
+                }
+                let msg = device_info.err();
+                if msg.is_some() {
+                    log::info!("{}", msg.unwrap());
+                }
+                std::thread::sleep(Duration::from_secs_f64(delay * i));
+                log::info!("Retrying..");
+                if i < 30f64 {i += 1f64;}
+            }
```

Comment:
Let's factor this out as a separate function, as it makes the loop and other state variables a bit cleaner if it can simply return the device in the success case.

For the sleep backoff, it feels inelegant to convert to/from floats, I would suggest something like this:

```rust
const MAX_SLEEP: Duration = Duration::from_secs(30);
const ONE_SECOND: Duration = Duration::from_secs(1);
let mut sleep = ONE_SECOND;

loop {
    ...
    std::thread::sleep(sleep);
    sleep = (sleep + ONE_SECOND).min(MAX_SLEEP);
}
```